orodha-keycloak/orodha_keycloak/connection.py

```diff
@@ -0,0 +1,162 @@
+                realm_name=realm_name,
+                client_id=client_id,
+                client_secret_key=client_secret_key,
+            )
+        else:
+            self.admin_connection = orodha_keycloak.connections.admin.create_admin_connection(
+                server_url=server_url,
+                realm_name=realm_name,
+                username=username,
+                password=password
+            )
+
+    def add_user(
+            self,
+            email: str,
+            username: str,
+            firstName: str,
+            lastName: str,
+            password: str
+    ):
+        """
+        Adds a user to keycloak with a password.
+
+        Args:
+                email(str): The email of the new user.
+                username(str): The username of the new user.
+                firstName(str): The first name of the new user.
+                lastName(str): The last name of the new user.
+                password(str): The password of the new user.
+
+        Returns:
+            new_user: The new user info genereated by the keycloak server.
+
+        """
+        new_user = self.admin_connection.create_user(
+            {
+                "email": email,
+                "username": username,
+                "enabled": True,
+                "firstName": firstName,
+                "lastName": lastName,
+                "credentials": [
+                    {
+                        "value": password,
+                        "type": "password",
+                    }
+                ],
+            },
+            exist_ok=False,
+        )
+
+        return new_user
+
+    def delete_user(self, user_id):
+        """
+        Deletes a keycloak user with a given user_id.
+
+        Args:
+            user_id(str): The user id of the user to be deleted.
+
+        Returns:
+            response: The response from the keycloak server with info about the deletion.
+        """
+        response = self.admin_connection.delete_user(user_id=user_id)
+
+        return response
+
+    def get_user(self, token: dict = None, user_id: str = None):
+        """
+        Takes either a user_id or a token and returns the user if they exist.
+
+        Args:
+            user_id(str): String user id of our user, is used to access keycloak in a query.
+            token(str): Our JWT token that we will use to decode and obtain the user from.
+
+        Returns:
+            user: The user, if any, that is associated with this user_identification value.
+        """
+
+        if token:
+            return_value = self.admin_connection.get_user(
+                self.decode_jwt(token).get("sub"))
+        else:
+            return_value = self.admin_connection.get_user(user_id)
+
+        return return_value
+
+    def decode_jwt(self, token):
+        """
+        Small helper function which decodes a JWT token using the client connection.
+
+        Args:
+            token(dict): A JWT token that we get from keycloak.
```

Comment:
@Edison-Stuart You'll want to have this be a string. Nearly all calls to this will be with a token extracted from a request header, ie.

``` headers = {"Authorization": "Bearer <this is the token>"}```

You'll want to be able to simply pass in the token. Also: the other method of this class that accepts a `token` argument treats it as a string; you'll want to maintain consistency here.